plugins/woocommerce-blocks/assets/js/blocks/checkout/inner-blocks/checkout-order-summary-block/edit.tsx

```diff
@@ -16,14 +21,37 @@ import {
 	getAllowedBlocks,
 } from '../../../cart-checkout-shared';
 import { OrderMetaSlotFill } from './slotfills';
+import { useContainerWidthContext } from '../../../../base/context';
+import { FormattedMonetaryAmount } from '../../../../../../packages/components';
 
 export const Edit = ( { clientId }: { clientId: string } ): JSX.Element => {
 	const blockProps = useBlockProps();
 	const { cartTotals } = useStoreCart();
 	const totalsCurrency = getCurrencyFromPriceResponse( cartTotals );
+	const totalPrice = parseInt( cartTotals.total_price, 10 );
 	const allowedBlocks = getAllowedBlocks(
 		innerBlockAreas.CHECKOUT_ORDER_SUMMARY
 	);
+	const { isSmall, isMobile } = useContainerWidthContext();
+	const [ isOpen, setIsOpen ] = useState( false );
+	const ariaControlsId = useId();
+
+	const orderSummaryProps =
+		isSmall || isMobile
+			? {
+					role: 'button',
+					onClick: () => setIsOpen( ! isOpen ),
+					'aria-expanded': isOpen,
+					'aria-controls': ariaControlsId,
+					tabIndex: 0,
+					onKeyDown: ( event: React.KeyboardEvent ) => {
+						if ( event.key === 'Enter' || event.key === ' ' ) {
+							setIsOpen( ! isOpen );
+						}
+					},
+			  }
+			: {};
```

Comment:
Instead of this, how about a block variation, like we have the empty cart and filled cart, could we have the collapsed order summary and expanded order summary?

Currently, if editing on mobile and you're trying to select the order summary block, it just toggles collapsed/expanded.

Maybe this is better as a follow up because not sure how common a use-case editing on mobile is, and the workaround is to use the list view to select it.

Reply:
Block variation is an interesting idea and sounds good to me. I would be happy to do it as a follow-up.